src/org/labkey/test/components/ui/grids/ManageGridViewsDialog.java

```diff
@@ -0,0 +1,174 @@
+    }
+
+    /**
+     * Check to see if a given view can be deleted.
+     *
+     * @param viewName Name of the view.
+     * @return True if the view can be deleted, false otherwise.
+     */
+    public boolean canViewBeDeleted(String viewName)
+    {
+        return Locator.tagWithClass("i", "fa-trash-o").findWhenNeeded(elementCache().viewRow(viewName)).isDisplayed();
+    }
+
+    /**
+     * Change the name of a view.
+     *
+     * @param currentName The current name of the view.
+     * @param newName The new name.
+     * @return This dialog.
+     */
+    public ManageGridViewsDialog changeViewName(String currentName, String newName)
+    {
+        Assert.assertTrue(String.format("View named '%s' is not editable.", currentName), canViewNameBeChanged(currentName));
+
+        Locator.tagWithClass("i", "fa-pencil").findElement(elementCache().viewRow(currentName)).click();
+
+        WebDriverWrapper.waitFor(()->elementCache().viewNameInput.isDisplayed(),
+                String.format("View name input for view '%s' did not show up in time.", currentName), 500);
+
+        Actions replaceCurrentText = new Actions(getDriver());
+        replaceCurrentText.sendKeys(Keys.END)
+                .keyDown(Keys.SHIFT)
+                .sendKeys(Keys.HOME)
+                .keyUp(Keys.SHIFT)
+                .sendKeys(newName)
+                .sendKeys(Keys.TAB)
+                .perform();
+
+        WebDriverWrapper.waitFor(()->!elementCache().viewNameInput.isDisplayed() && getViewNames().contains(newName),
+                String.format("New view name '%s' was not added to the list of views.", newName), 1_500);
+
+        return this;
+    }
+
+    /**
+     * Delete the given view.
+     *
+     * @param viewName The name of the view to delete.
+     * @return This dialog.
+     */
+    public ManageGridViewsDialog deleteView(String viewName)
+    {
+        Assert.assertTrue(String.format("View named '%s' cannot be deleted.", viewName), canViewBeDeleted(viewName));
+
+        Locator.tagWithClass("i", "fa-trash-o").findWhenNeeded(elementCache().viewRow(viewName)).click();
+
+        return this;
+    }
+
+    /**
+     * Get the confirmation text asking if you really want to delete this view.
+     *
+     * @return The confirmation text.
+     */
+    public String getDeleteConfirmationText()
+    {
+        return elementCache().deleteConfirmText.getText();
+    }
+
+    /**
+     * Click the 'Yes' button when asked to delete a button.
+     *
+     * @return This dialog.
+     */
+    public ManageGridViewsDialog clickDeleteYesButton()
+    {
+        elementCache().deleteYesButton.click();
+        return this;
+    }
+
+    /**
+     * Click 'No' button to not delete a button.
+     *
+     * @return This dialog.
+     */
+    public ManageGridViewsDialog clickDeleteNoButton()
+    {
+        elementCache().deleteNoButton.click();
+        return this;
+    }
```

Comment:
These method names are confusing. What about confirm and cancel
```suggestion
    public ManageGridViewsDialog confirmDelete()
    {
        elementCache().deleteYesButton.click();
        return this;
    }

    /**
     * Click 'No' button to not delete a button.
     *
     * @return This dialog.
     */
    public ManageGridViewsDialog cancelDelete()
    {
        elementCache().deleteNoButton.click();
        return this;
    }
```

Reply:
Changed.